python/Ganga/GPIDev/Lib/File/MassStorageFile.py

```diff
@@ -198,20 +202,52 @@ def getWNScriptDownloadCommand(self, indent):
 
         return script
 
+    def _mkdir(self, massStoragePath, exitIfNotExist=False):
+        """
+        Creates a folder on the mass Storage corresponding to the given path
+        Args:
+            massStoragePath (str): This is the path we want to make if it doesn't exist.
+        """
+
+        massStorageConfig = getConfig('Output')['MassStorageFile']['uploadOptions']
+        mkdir_cmd = massStorageConfig['mkdir_cmd']
+        ls_cmd = massStorageConfig['ls_cmd']
+
+        # create the last directory (if not exist) from the config path
+        pathToDirName = os.path.dirname(massStoragePath)
+        dirName = os.path.basename(massStoragePath)
+
+        directoryExists = False
+
+        (exitcode, mystdout, mystderr) = self.execSyscmdSubprocess('%s %s' % (ls_cmd, quote(pathToDirName)))
+        if exitcode != 0:
+            if exitIfNotExist:
+                self.handleUploadFailure(mystderr, '1) %s %s' % (ls_cmd, pathToDirName))
+                raise GangaException(mystderr)
+            directoryExists = False
```

Comment:
This should be just:
```
if exitcode != 0 and exitIfNotExist:
    self.handleUploadFailure...
```
The additional `directoryExists = False` isn't needed anymore.